Makefile

```diff
@@ -230,13 +230,13 @@ check test: samtools $(BGZIP) $(TEST_PROGRAMS)
 
 
 test/merge/test_bam_translate: test/merge/test_bam_translate.o test/test.o libst.a $(HTSLIB)
-	$(CC) $(ALL_LDFLAGS) -o $@ test/merge/test_bam_translate.o test/test.o libst.a $(HTSLIB_LIB) $(ALL_LIBS) -lpthread
+	$(CC) $(ALL_LDFLAGS) -o $@ test/merge/test_bam_translate.o test/test.o libst.a tmp_file.o $(LZ4OBJS) $(HTSLIB_LIB) $(ALL_LIBS) -lpthread
```

Comment:
Instead of adding these here, just move them from `AOBJS` to `LIBST_OBJS` — sharing object files between samtools and these test programs is exactly what _libst.a_ is for.